Minio/DataModel/ObjectOperationArgs.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Security.Cryptography;
+using System.Xml.Serialization;
+using Minio.DataModel;
+using RestSharp;
+
+
+namespace Minio
+{
+    public class ObjectArgs : BucketArgs
+    {
+        internal string ObjectName { get; set; }
+
+        public ObjectArgs(string bucket, string obj)
+                    : base(bucket)
+        {
+            this.ObjectName = obj;
+        }
+
+    }
+
+    public class GetObjectLegalHoldArgs : ObjectArgs
+    {
+        internal string VersionId { get; set; }
+
+        public GetObjectLegalHoldArgs(string bucket, string obj)
+                        : base(bucket, obj)
+        {
+        }
+        public new void Validate()
+        {
+            utils.ValidateBucketName(this.BucketName);
+            utils.ValidateObjectName(this.ObjectName);
+        }
+        public ObjectArgs WithVersionID(string v)
```

Comment:
this is the same issue as earlier, this cannot return objectArgs, otherwise this is not even required

Reply:
Yes.